tests/sentry/backup/test_correctness.py

```diff
@@ -0,0 +1,188 @@
+import io
+import os
+from pathlib import Path
+from typing import Dict, List, NewType, Optional, Protocol, Tuple
+
+from dateutil import parser
+from django.db import IntegrityError
+from jsondiff import diff
+from pydantic import FilePath, PositiveInt
+
+from sentry.db.postgres.roles import in_test_psql_role_override
+from sentry.runner.commands.backup import exporter, importer
+from sentry.utils import json
+from sentry.utils.json import JSONData
+from sentry.utils.pytest.fixtures import django_db_all
+
+ComparatorName = NewType("ComparatorName", str)
+ModelName = NewType("ModelName", str)
+
+
+class InstanceID:
+    """Every entry in the generated backup JSON file should have a unique model+pk combination, which serves as its identifier."""
+
+    model: ModelName
+    pk: PositiveInt
+
+    def __init__(self, model: ModelName, pk: PositiveInt):
+        self.model = model
+        self.pk = pk
+
+    def __eq__(self, other):
+        return (
+            hasattr(other, "model")
+            and self.model == other.model
+            and hasattr(other, "pk")
+            and self.pk == other.pk
+        )
+
+    def __hash__(self):
+        return hash((self.model, self.pk))
+
+    def __str__(self):
+        return f'InstanceID(model: "{self.model}", pk: {self.pk})'
```

Comment:
this looks more like a `__repr__` than a `__str__`

Reply:
I tried `__repr__` to begin with, but it interacted poorly with the debugger (especially the multiline versions below), so I decided to go with string instead.

Reply:
`__str__` is meant for conversion to string -- these are not conversions these are programmer representations so they should use `__repr__`.

Reply:
Added a comment with my debugger-based justification for using `__str__` to the code, which I think is okay for now.

Reply:
please stop resolving this.  you haven't actually shown the problem and this is not correct

Reply:
Changed all of the `__str__()` to `print()` and call the method directly. Does that work?

Reply:
I want to give you the benefit of the doubt but you have not shown your actual problem still after being prompted for it and I'm getting a little frustrated.  please show me what your actual problem is instead of continuing to break the object model

Reply:
I'm sorry to frustrate you. I've attached an image of the issue:

![Screenshot 2023-07-06 at 10 43 25](https://github.com/getsentry/sentry/assets/3709945/ca817dde-4463-4bd1-8c47-60b7e3fac9c1)

The highlighted line in the debugger pane in the top-left of the view is one example of pretty-printed output (which would be desirable in actual logs and test reports) being difficult to read in the debugger context.

I'm not sure I understand how the latest change breaks the object model, since afaict a `print()` _method_ on a class is not considered "special" like `__repr__()` or `__str__()`, it's just a method I can call like any other, right? But in any case, just let me know what your preferred approach to fixing this issue would be and I'll do it.

Reply:
Would you be okay with me using a custom method (named whatever you prefer) for doing the newline-ful pretty printing, and using it the same way I use the method currently called `print()` in the current iteration? Ie, doing the manual assertion formatting you suggest via a method call?

Reply:
Is the new `pretty()` method on the relevant classes acceptable as a form of "manual" formatting of my assertion messages?